### PR TITLE
Add region name property to clients

### PR DESCRIPTION
**Pull request by jamesls**
As part of this change, I removed the `clone_client` method.  This ended up not being as useful as I thought, and gets harder to maintain as we add new features to the client.

For now, the region_name is just copied out from the endpoint.  I think eventually we're moving region name out of endpoint, but we can still maintain this interface.  That way, this should unblock any customizations that get the region name from the endpoint.  They can now just grab this value from clients.

cc @kyleknap @danielgtaylor

### CODE REVIEW

File: tests/unit/test_endpoint.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Copyright 2012-2014 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# Copyright 2012-2015 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (danielgtaylor) [inline tests/unit/test_endpoint.py]: It doesn't look like there are any other changes to this file. Any reason to update the copyright year?

**Comment by danielgtaylor** [comment]
LGTM and I'm :+1: on removing `clone_client` after talking with @kyleknap about it. :ship:-it!

File: tests/unit/test_endpoint.py (hunk 1 above)
Comment (jamesls) [inline tests/unit/test_endpoint.py]: This was part of a larger refactoring, which I rebased.  Looks like I didn't quite get them split cleanly, I'll change this back.

**Comment by kyleknap** [comment]
Looks good to me. Its fine that we are using the ``endpoint.region_name`` as that is where ``region_name`` should be sourced from in the client until we move the ``region_name`` logic out of the endpoint and into the client. :ship:

**Pull request merged**